src/main/java/seedu/address/model/person/insurance/claim/Claim.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.model.person.insurance.claim;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * An object that represents a claim that the client has made under an insurance plan.
+ */
+public class Claim {
+    public static final String INVALID_CLAIM_ID = "Invalid Claim ID, must follow: "
+            + "Letter + 4 digits only.";
+
+    public static final String NEGATIVE_CLAIM_AMOUNT = "Claim Amount cannot be negative";
+    public static final String INVALID_CLAIM_AMOUNT = "Claim Amount must be a positive integer";
+
+    private final String claimId;
+    private final boolean claimStatus;
+    private final int claimAmount;
+
+    /**
+     * Constructs a claim object based on the input parameters.
+     *
+     * @param claimId     claim id obtained through official channels.
+     * @param claimStatus current status of the claim, only takes a boolean value.
+     * @param claimAmount claim amount recorded in cents, can only be a positive integer.
+     */
+    public Claim(String claimId, boolean claimStatus, int claimAmount) {
+        this.claimId = claimId;
+        this.claimStatus = claimStatus;
+        this.claimAmount = claimAmount;
+    }
+
+    /**
+     * Returns the {@code claimId} of this claim object.
+     */
+    public String getClaimId() {
+        return this.claimId;
+    }
+
+    /**
+     * Returns the {@code claimStatus} of this claim object.
+     */
+    public boolean getClaimStatus() {
+        return this.claimStatus;
+    }
+
+    /**
+     * Returns the {@code claimAmount} of this claim object.
+     */
+    public int getClaimAmount() {
+        return this.claimAmount;
+    }
+
+    /**
+     * Checks if the claim id is valid based on preset notations.
+     * Regular expression pattern:
+     * ^ - start of the string
+     * [A-Za-z] - any uppercase or lowercase letter
+     * \d{4} - exactly 4 digits
+     * $ - end of the string
+     * This method was created with the aid of ChatGPT.
+     *
+     * @param claimId the claimId string to be checked for validity.
+     * @throws ParseException if the claimId is invalid.
+     */
+    public static void checkValidClaimId(String claimId) throws ParseException {
+        String pattern = "^[A-Za-z]\\d{4}$";
```

Comment:
I am unsure if we should assume the claim id to have a regular expression pattern. I was thinking if claim id expression would be different for different insurance companies

Reply:
I think it is good to define and restrict to prevent outrageous inputs that may break the code. Lets keep it as is for now and ensure to define it in the UG.